File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeUseMvoStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class AttributeUseMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'attributeSetAttributeUseIdAttributeSetId' => 'string',
            'attributeSetAttributeUseIdAttributeId' => 'string',
            'attributeSetVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeUseMvoStateEventId
     */
    private $value;

    /**
     * @param AttributeUseMvoStateEventId $value
     */
    public function __construct(AttributeUseMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeUseMvoStateEventId();
        }
    }

    /**
     * @return AttributeUseMvoStateEventId
     */
    public function toAttributeUseMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getAttributeSetAttributeUseIdAttributeSetId()
    {
        return $this->value->getAttributeSetAttributeUseId()->getAttributeSetId();
    }

    /**
     * @param string $attributeSetAttributeUseIdAttributeSetId
     */
    public function setAttributeSetAttributeUseIdAttributeSetId($attributeSetAttributeUseIdAttributeSetId)
    {
        $this->value->getAttributeSetAttributeUseId()->setAttributeSetId($attributeSetAttributeUseIdAttributeSetId);
    }

    /**
     * @return string
     */
    public function getAttributeSetAttributeUseIdAttributeId()
    {
        return $this->value->getAttributeSetAttributeUseId()->getAttributeId();
    }

    /**
     * @param string $attributeSetAttributeUseIdAttributeId
     */
    public function setAttributeSetAttributeUseIdAttributeId($attributeSetAttributeUseIdAttributeId)
    {
        $this->value->getAttributeSetAttributeUseId()->setAttributeId($attributeSetAttributeUseIdAttributeId);
    }

    /**
     * @return Long
     */
    public function getAttributeSetVersion()
    {
        return $this->value->getAttributeSetVersion();
    }

    /**
     * @param Long $attributeSetVersion
     */
    public function setAttributeSetVersion($attributeSetVersion)
    {
        $this->value->setAttributeSetVersion($attributeSetVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getAttributeSetAttributeUseIdAttributeSetId(),
            $this->getAttributeSetAttributeUseIdAttributeId(),
            $this->getAttributeSetVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return AttributeUseMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setAttributeSetAttributeUseIdAttributeSetId($pValues[0]);
        $this->setAttributeSetAttributeUseIdAttributeId($pValues[1]);
        $this->setAttributeSetVersion($pValues[2]);
        return $this;
    }

}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AttributeUseMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeUseMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "attributeSetAttributeUseIdAttributeSetId",
            "attributeSetAttributeUseIdAttributeId",
            "attributeSetVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeUseMvoStateEventId value;

    public AttributeUseMvoStateEventIdFlattenedDto()
    {
        this(new AttributeUseMvoStateEventId());
    }

    public AttributeUseMvoStateEventIdFlattenedDto(AttributeUseMvoStateEventId value)
    {
        this.value = value;
    }

    public AttributeUseMvoStateEventId toAttributeUseMvoStateEventId()
    {
        return this.value;
    }


    public String getAttributeSetAttributeUseIdAttributeSetId()
    {
        return this.value.getAttributeSetAttributeUseId().getAttributeSetId();
    }

    public void setAttributeSetAttributeUseIdAttributeSetId(String attributeSetAttributeUseIdAttributeSetId)
    {
        this.value.getAttributeSetAttributeUseId().setAttributeSetId(attributeSetAttributeUseIdAttributeSetId);
    }

    public String getAttributeSetAttributeUseIdAttributeId()
    {
        return this.value.getAttributeSetAttributeUseId().getAttributeId();
    }

    public void setAttributeSetAttributeUseIdAttributeId(String attributeSetAttributeUseIdAttributeId)
    {
        this.value.getAttributeSetAttributeUseId().setAttributeId(attributeSetAttributeUseIdAttributeId);
    }

    public Long getAttributeSetVersion()
    {
        return this.value.getAttributeSetVersion();
    }

    public void setAttributeSetVersion(Long attributeSetVersion)
    {
        this.value.setAttributeSetVersion(attributeSetVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeUseMvoStateEventIdFlattenedDto other = (AttributeUseMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

